buildSrc/src/main/java/com/osacky/flank/gradle/SanityConfigValidation.kt

```diff
@@ -0,0 +1,32 @@
+package com.osacky.flank.gradle
+
+import org.gradle.api.GradleException
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+
+@Throws(GradleException::class)
+fun checkIfSanityAndValidateConfigs(config: FladleConfig) = when (config) {
+  is FlankGradleExtension -> config.checkAndValidateConfig() { option, _ ->
+    "Incorrect [base] configuration. [$option] can't be used together with sanityRobo."
+  }
+  is FladleConfigImpl -> config.checkAndValidateConfig(config.name) { option, name ->
+    "Incorrect [$name] configuration. [$option] can't be used together with sanityRobo. " +
+      "If you want to launch robo test run without robo script place only clearPropertiesForSanityRobo() into [$name] configuration"
+  }
+  else -> throw GradleException("Unable to check for sanity, check config type")
```

Comment:
Let's throw an `IllegalStateException` and add the `config`'s class name (or just `toString()` to the message.
This will help with debugging down the line.